libr/bin/bin.c

```diff
@@ -1219,20 +1207,28 @@ R_API void r_bin_bind(RBin *bin, RBinBind *b) {
 	}
 }
 
-R_API RBuffer *r_bin_create(RBin *bin, const ut8 *code, int codelen,
-			     const ut8 *data, int datalen) {
-	RBinFile *a = r_bin_cur (bin);
-	RBinPlugin *plugin = r_bin_file_cur_plugin (a);
+R_API RBuffer *r_bin_create(RBin *bin, const char *plugin_name,
+	const ut8 *code, int codelen, const ut8 *data, int datalen,
+	RBinArchOptions *opt) {
+
+	r_return_val_if_fail (bin && plugin_name && opt, NULL);
+
+	RBinPlugin *plugin = r_bin_get_binplugin_by_name (bin, plugin_name);
+	if (!plugin) {
+		R_LOG_WARN ("Cannot find RBin plugin named '%s'.\n", plugin_name);
+		return NULL;
+	} else if (!plugin->create) {
```

Comment:
dont return after else

Reply:
I'm just `else` after return, I'm `else if`, which is different. I'm just dealing with the possible things that could happen..
1) plugin may be NULL
OR
2) if plugin was not NULL, plugin->create may be null

If I don't use the `else if` I'd just have to put the `if` on the following line, losing a bit of meaning to the mentioned code.

Reply:
the behaviour is the  same and its clear what it does. anything after return